### PR TITLE
Update percentage of users able to use browser notifications to 40%

### PR DESCRIPTION
**Pull request by mattsherman**
This PR increases the number of users able to use browser notifications to 40%, while keeping the number of users nudged on other pages with the notice at the top of the screen to 5%.

To test, use a browser environment that supports Service Workers for `HTTP` connections (like Firefox with the developer toolbox open) and run calypso locally like so:

```
DISABLE_FEATURES=push-notifications ENABLE_FEATURES=push-notifications-ab-test make run
```

This will ensure that you have the configuration flags set as they will be currently in production.

Then, use the environment badge in the bottom right of Calypso to set the AB test variations and verify correct behavior.

`browserNotifications` | `browserNotificationsPreferences` | Nudge visible? | Preferences visible?
---------------------| --------------------------------- | -------------- | -------------------
`disabled` | `disabled` | no | no
`enabled` | `disabled` | yes | yes
`enabled` | `enabled` | yes | yes
`disabled` | `enabled` | no | yes

Test live: https://calypso.live/?branch=update/web-push-abtest-increase

### CODE REVIEW

File: client/boot/index.js [hunk 1]
```diff
@@ -211,7 +211,8 @@ function reduxStoreReady( reduxStore ) {
 		reduxStore.dispatch( setCurrentUserId( user.get().ID ) );
 		reduxStore.dispatch( setCurrentUserFlags( user.get().meta.data.flags.active_flags ) );
 
-		const participantInPushNotificationsAbTest = config.isEnabled('push-notifications-ab-test') && abtest('browserNotifications') === 'enabled';
+		const participantInPushNotificationsAbTest = config.isEnabled( 'push-notifications-ab-test' ) &&
```
Comment (mjuhasz) [inline client/boot/index.js]: What we want to express is that

- if the push notification AB test is turned on, AND
- the user is eligible in EITHER of the active tests

then we consider the user participant.

Logical AND has higher precedence than Logical OR so expressing the intent I described, grouping is necessary here:

```
		const participantInPushNotificationsAbTest = config.isEnabled( 'push-notifications-ab-test' ) &&
			( abtest( 'browserNotifications' ) === 'enabled' || abtest( 'browserNotificationsPreferences' ) === 'enabled' );
```

**Comment by mjuhasz** [comment]
I tested this branch, and works as expected.

To fully understand why it works the way it does, [this line](https://github.com/Automattic/wp-calypso/blob/master/client/layout/index.jsx#L95) is key.

Right now the behavior using this branch is correct as is, but I'd like to address the comment I made before merging, so I'm not giving a thumbs up yet. Turning off the `push-notifications-ab-test` flag would not turn off the push notification feature if this branch was merged.

File: client/boot/index.js (hunk 1 above)
Comment (mattsherman) [inline client/boot/index.js]: Good catch... fixed!

File: client/lib/abtest/active-tests.js [hunk 2]
```diff
@@ -8,6 +15,21 @@ module.exports = {
...
+	// `browserNotificationsPreferences` controls whether or not users see the
+	// preference to enable browser notifications in /me/notifications;
+	// any users with `browserNotifications` enabled will also see the preference;
+	// this is a temporary test just to allow us to ramp up the load gradually so
+	// that make sure things will be smooth when we go to launching browser
+	// notifications for 100% of users
+	browserNotificationsPreferences: {
+		datestamp: '20160801',
+		variations: {
+			disabled: 0,
```
Comment (mjuhasz) [inline client/lib/abtest/active-tests.js]: I am not sure how `0` would be handled here - It'd be more clear if we wrote 60: then it'd be expressed as percentage (60%-40%). Even though allocations do not have to add up to 100, it's just easier to read that way.

Comment (mattsherman) [inline client/lib/abtest/active-tests.js]: Oh wow... don't know what I was thinking here... certainly! Fixed! (I'm pretty sure that would have enabled things for 100% of users in this test.)

**Comment by mjuhasz** [comment]
Thanks for the fixes! This one is good to go 👍

**Pull request merged**